README.md

```diff
@@ -135,6 +135,18 @@ var limited = qs.parse('a=b&c=d', { parameterLimit: 1 });
 assert.deepEqual(limited, { a: 'b' });
 ```
 
+If you want an error to be thrown whenever the parameter limit is exceeded, set the `throwOnLimitExceeded` option to `true`. This option will generate a descriptive error if the query string exceeds the parameterLimit.
+```javascript
+try {
+    qs.parse('a=1&b=2&c=3&d=4', { parameterLimit: 3, throwOnLimitExceeded: true });
+} catch (err) {
+    assert(err instanceof Error);
+    assert.strictEqual(err.message, 'Parameter limit exceeded. Only 3 parameters allowed.');
+}
+```
+
+When `throwOnLimitExceeded` is set to `false` (default), **qs** will parse up to the specified `parameterLimit` and ignore the rest without throwing an error.
```

Comment:
account for arrayLimit here also

Reply:
A question, currently in case `arrayLimit` is exceeded, it is not getting _silently truncated_, right? And we are just throwing the error in case it does?

For example, If we were to account for `arrayLimit` here, the following test case has to pass,

```js
{
        description: 'throwOnLimitExceeded is set to false, parse upto specified limit and ignore the rest',
        query: 'a[]=1&a[]=2&a[]=3&a[]=4',
        options: { arrayLimit: 5, throwOnLimitExceeded: false },
        expected: { a: ['1', '2', '3'] }
 }
````

But it doesn't, on a limited test run (not a factor), I get,

```console
Test Case: throwOnLimitExceeded is set to false, parse upto specified limit and ignore the rest
Query: a[]=1&a[]=2&a[]=3&a[]=4
Options: { arrayLimit: 5, throwOnLimitExceeded: false }
Parsed Result: { a: [ '1', '2', '3', '4' ] }
Assertion failed: Expected {"a":["1","2","3"]} but got {"a":["1","2","3","4"]}
```

Reply:
An arrayLimit of 5 would require 6 items to exceed it; you've only got 4.

Reply:
Okay, that was a wrong test case, I am not going to prolong that. 

However, I believe array is not truncated to configured limit. In order to test that, I had checked out to main branch and added this test case to the `parse` test cases,

```javascript
t.test('silently truncates array to configured limit', function (sst) {
    const result = qs.parse('a[]=1&a[]=2&a[]=3&a[]=4&a[]=5&a[]=6&a[]=7', { arrayLimit: 5 });

    sst.deepEqual(result, {a: [1, 2, 3, 4, 5] }, 'Should parse and truncate silently');
    sst.end();
});
```

We expect it to pass, but it fails.
```console
# silently truncates array to configured limit
not ok 255 Should parse and truncate silently
  ---
    operator: deepEqual
    expected: |-
      { a: [ 1, 2, 3, 4, 5 ] }
    actual: |-
      { a: [ '1', '2', '3', '4', '5', '6', '7' ] }
```

Reply:
Do you think it needs to be addressed? I can open a new issue and get started on this.

Reply:
Do you think it needs to be addressed? I can open a new issue and get started on this.

Reply:
ah, sorry for the confusion. arrayLimit applies with explicit indexes. eg, `qs.parse('a[0]=1&a[10]=2', { arrayLimit: 5 })`.

Reply:
I have updated the test case to include explicit indices,

```js
t.test('silently truncates array to configured limit', function (sst) {
    const result = qs.parse('a[0]=1&a[1]=2&a[2]=3&a[3]]=4&a[4]=5&a[5]=6&a[6]=7', { arrayLimit: 5 });

    sst.deepEqual(result, {a: [1, 2, 3, 4, 5] }, 'Should parse and truncate silently');
    sst.end();
});
```

I am getting this, 

```console
# silently truncates array to configured limit
not ok 255 Should parse and truncate silently
  ---
    operator: deepEqual
    expected: |-
      { a: [ 1, 2, 3, 4, 5 ] }
    actual: |-
      { a: { 0: '1', 1: '2', 2: '3', 3: '4', 4: '5', 5: '6', 6: '7' } }
```

It seems all seven elements of `a` are getting parsed.

Reply:
right - but the output is an object instead of an array.

the option would cause it to throw when it would fail to produce an array.

Reply:
Okay

Reply:
> qs will also limit specifying indices in an array to a maximum index of 20. Any array members with an index of greater than 20 will instead be converted to an object with the index as the key. This is needed to handle cases when someone sent, for example, a[999999999] and it will take significant time to iterate over this huge array.
[qs README](https://github.com/ljharb/qs/blob/main/README.md#parsing-arrays)

Like `parameter` it would not truncate the array silently, rather it creates an object like we have been seeing in above test example. And if the provided array is within limit, then it parses the array in expected output.

So, instead of adding about `throwOnLimitExceeded` effect on `arrayLimit` here, it would be better to add an example in the [parsing-arrays](https://github.com/ljharb/qs/blob/main/README.md#parsing-arrays) section.